platform.py

```
@@ -19,234 +19,477 @@
+            # Version matches, use tool
+            self.packages[tool_name]["version"] = paths['tool_path']
+            self.packages[tool_name]["optional"] = False
+            logger.debug(f"Tool {tool_name} found with correct version")
+            return True
+        else:
+            # Wrong version, reinstall
+            logger.info(f"Reinstalling {tool_name} due to version mismatch")
+            safe_remove_directory(paths['tool_path'])
+            return self.install_tool(tool_name, retry_count + 1)
+
+    @lru_cache(maxsize=1)
```

Ruff (0.11.9)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)